### PR TITLE
feat(options): add `defaultEditor` option

### PR DESCRIPTION
**Pull request by mrtnvgr**
This PR adds `defaultEditor` option from home-manager's `neovim` module

### CODE REVIEW

**Comment by traxys** [comment]
I don't think this should be added in this place, as it is a HM specific option. I think this has more it's place in the nixvim/wrappers/hm.nix file

File: wrappers/modules/output.nix [hunk 1]
```diff
@@ -43,6 +43,14 @@ in {
       '';
     };
 
+    defaultEditor = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Configure <command>nvim</command> as the default editor using the EDITOR environment variable
```
Comment (GaetanLepage) [inline wrappers/modules/output.nix:50]: Shouldn't we simply use `nvim` here (because the doc is generated from markdown) ?

**Comment by traxys** [comment]
I feel like the option definition itself should be in the hm file (and maybe nixos) because it makes no sense for the standalone module. Something like programs.nixvim.defaultEditor should work ?

**Comment by mrtnvgr** [comment]
> I feel like the option definition itself should be in the hm file (and maybe nixos) because it makes no sense for the standalone module. Something like programs.nixvim.defaultEditor should work ?

I don't quite understand how can I do that

**Comment by traxys** [comment]
```nix
    programs.nixvim = mkOption {
      default = {};
      type = types.submodule ([
          {
            options = {
              enable = mkEnableOption "nixvim";
              defaultEditor = mkEnableOption "Set nixvim as the default option";
            };
          }
        ]
        ++ shared.topLevelModules);
    };
```

Something like that should work (not tested)

**Review by traxys — APPROVED** [review]

**Pull request merged**